include/lsst/afw/image/PhotoCalib.h

```diff
@@ -0,0 +1,369 @@
+ */
+
+#include "lsst/afw/math/BoundedField.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/table/Source.h"
+
+namespace lsst { namespace afw { namespace image {
+
+class PhotoCalib {
+public:
+    PhotoCalib(PhotoCalib const &) = delete;
+    PhotoCalib(PhotoCalib &&) = delete;
+    PhotoCalib & operator=(PhotoCalib const &) = delete;
+    PhotoCalib & operator=(PhotoCalib &&) = delete;
+
+    /**
+     * Create a empty, zeroed calibration.
+     */
+    PhotoCalib(): _fluxMag0(0.0), _fluxMag0Sigma(0.0) {}
+
+    /**
+     * Create a non-spatially-varying calibration.
+     *
+     * @param[in]  fluxMag0       The constant flux/magnitude zero point (counts at magnitude 0).
+     * @param[in]  fluxMag0Sigma  The error on the zero point.
+     */
+    explicit PhotoCalib(double fluxMag0, double fluxMag0Sigma=0):
+        _fluxMag0(fluxMag0), _fluxMag0Sigma(fluxMag0Sigma) {}
+
+
+    /**
+     * Create a spatially-varying calibration.
+     *
+     * @param[in]  zeroPoint      The spatially varying photometric zero point.
+     * @param[in]  fluxMag0Sigma  The error on the zero point.
+     */
+    PhotoCalib(std::shared_ptr<afw::math::BoundedField> zeroPoint, double fluxMag0Sigma=0):
+    _zeroPoint(zeroPoint), _fluxMag0(computeFluxMag0(zeroPoint)), _fluxMag0Sigma(fluxMag0Sigma) {}
+
+    //@{
+    /**
+     * Convert counts in ADU to maggies.
+     *
+     * If passed point, use the exact calculation at that point, otherwise, use the mean scaling factor.
+     *
+     * @param[in]  counts  The source flux in ADU.
+     * @param[in]  point   The point that flux is measured at (must be within the domain of the
+     *                     BoundedField of this PhotoCalib).
+     * @param[in]  throwOnNegativeFlux  Raise an exception when passed negative counts.
+     *
+     * @return     The flux in maggies.
+     */
+    double countsToMaggies(double counts,
+                           const afw::geom::Point<double, 2> &point,
+                           bool throwOnNegativeFlux=false) const;
+    double countsToMaggies(double counts,
+                           bool throwOnNegativeFlux=false) const;
+    //@}
+
+    //@{}
+    /**
+     * Convert counts and error in counts (ADU) to maggies and maggies error.
+     *
+     * If passed point, use the exact calculation at that point, otherwise, use the mean scaling factor.
+     *
+     * @param[in]  counts       The source flux in ADU.
+     * @param[in]  countsSigma  The counts error (sigma).
+     * @param[in]  point        The point that flux is measured at (must be within the domain of the
+     *                          BoundedField of this PhotoCalib).
+     * @param[in]  throwOnNegativeFlux  Raise an exception when passed negative counts.
+     *
+     * @return     The flux in maggies and error (sigma).
+     */
+    std::pair<double, double> countsToMaggies(double counts, double countsSigma,
```

Comment:
Pair is a bit generic, and it can be hard to remember what the first and second elements are after it's been passed around some. But the appropriateness/inappropriateness of tuples as return types is a controversial subject, so I won't belabor the point.

I'll have to think about the overloads; part of the problem is that apart from `Point` vs. average I don't see much of a pattern to them.